src/Swarm/Game/Step.hs

```diff
@@ -1671,13 +1670,17 @@ execConst c vs s k = do
     RID ->
     m Robot
   getRobotWithinTouch rid = do
-    mother <- robotWithID rid
-    other <- mother `isJustOrFail` ["There is no robot with ID", from (show rid) <> "."]
-    -- Make sure it is in the same location
-    loc <- use robotLocation
-    ((other ^. robotLocation) `manhattan` loc <= 1)
-      `holdsOrFail` ["The robot with ID", from (show rid), "is not close enough."]
-    return other
+    cid <- use robotID
+    if cid == rid
+      then get @Robot
```

Comment:
I don't quite understand why this didn't work before - isn't a robot within Manhattan distance 1 of itself?  But this code is certainly sensible.

Reply:
Hah, not if it is not inside the robot map. :sweat_smile:

Reply:
Ahhh, of course!